executor/common_linux.h

```diff
@@ -1485,6 +1485,215 @@ static long syz_io_uring_submit(volatile long a0, volatile long a1, volatile lon
 
 #endif
 
+#if SYZ_EXECUTOR || __NR_syz_btf_id_by_name
+
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+// Some items in linux/btf.h are relatively new, so we copy them here for
+// backward compatibility.
+#define BTF_MAGIC 0xeB9F
+
+struct btf_header {
+	__u16 magic;
+	__u8 version;
+	__u8 flags;
+	__u32 hdr_len;
+	__u32 type_off;
+	__u32 type_len;
+	__u32 str_off;
+	__u32 str_len;
+};
+
+#define BTF_INFO_KIND(info) (((info) >> 24) & 0x0f)
+#define BTF_INFO_VLEN(info) ((info)&0xffff)
+
+#define BTF_KIND_INT 1
+#define BTF_KIND_ARRAY 3
+#define BTF_KIND_STRUCT 4
+#define BTF_KIND_UNION 5
+#define BTF_KIND_ENUM 6
+#define BTF_KIND_FUNC_PROTO 13
+#define BTF_KIND_VAR 14
+#define BTF_KIND_DATASEC 15
+
+struct btf_type {
+	__u32 name_off;
+	__u32 info;
+	union {
+		__u32 size;
+		__u32 type;
+	};
+};
+
+struct btf_enum {
+	__u32 name_off;
+	__s32 val;
+};
+
+struct btf_array {
+	__u32 type;
+	__u32 index_type;
+	__u32 nelems;
+};
+
+struct btf_member {
+	__u32 name_off;
+	__u32 type;
+	__u32 offset;
+};
+
+struct btf_param {
+	__u32 name_off;
+	__u32 type;
+};
+
+struct btf_var {
+	__u32 linkage;
+};
+
+struct btf_var_secinfo {
+	__u32 type;
+	__u32 offset;
+	__u32 size;
+};
+
+// Set the fixed mapping address to be 64 pages after the data section, which
+// ends in SYS_DATA_OFFSET + (SYZ_NUM_PAGES + 1) * SYS_PAGE_SIZE
+#define BTF_VMLINUX_FIX_ADDRESS ((char*)(SYZ_DATA_OFFSET + (SYZ_NUM_PAGES + 1 + 64) * SYZ_PAGE_SIZE))
```

Comment:
Here is an issue I am facing now: What is the correct way to include these constants?

What I tried:
```c
#ifndef SYZ_EXECUTOR
#include "defs.h"
#endif
```
and then add `defs.h` inside `executorFilenames` in `pkg/csource/gen.go`. But then `int64_t` is replaced to `int64_t_t`.

Reply:
Or maybe simply using a static array with a reasonable bound size is not a bad idea? If we use a fixed address here, other people writing pseudo-syscalls in the future will need to check this address every time if they are also doing a `mmap` with a fixed address.

Reply:
These constants can't be used here. common*.h headers go into C reproducers, defs.h is not part of C reproducers.
If we have a reasonable upper bound on memory size, a global variable will be simpler, yes.